march_shared_classes/src/march_shared_classes/gait/subgait.py

```diff
@@ -229,6 +235,37 @@ def create_interpolated_setpoints(self, timestamps):
 
             joint.setpoints = new_joint_setpoints
 
+    @classmethod
+    def interpolate_subgaits(cls, base_subgait, other_subgait, parameter):
+        if parameter == 1:
+            return other_subgait
+        if parameter == 0:
+            return base_subgait
+        if not (0 < parameter < 1):
+            raise ValueError('Parameter for interpolation should be in the interval [0, 1], but is {0}'
+                             .format(parameter))
+
+        if len(base_subgait.joints) != len(other_subgait.joints):
```

Comment:
wouldn't it be better to:
```suggestion
        if base_subgait.joints == other_subgait.joints:
```
or could a possible situation be that the joints are not in the same order, in that case:
```suggestion
        if sorted(base_subgait.joints) == sorted(other_subgait.joints):
```